testing/tests/DevExpress.ui.widgets/listParts/commonTests.js

```diff
@@ -1286,6 +1302,275 @@ QUnit.module('options changed', moduleSetup, () => {
+
+        $selectionControls = $(instance.element()).find(`.${LIST_SELECT_CHECKBOX_CLASS}`);
+        assert.strictEqual($selectionControls.length, 2);
+
+        instance.option('showSelectionControls', false);
+        $selectionControls = $(instance.element()).find(`.${LIST_SELECT_CHECKBOX_CLASS}`);
+        assert.strictEqual($selectionControls.length, 0);
+
+        instance.option('showSelectionControls', true);
+        $selectionControls = $(instance.element()).find(`.${LIST_SELECT_CHECKBOX_CLASS}`);
+        assert.strictEqual($selectionControls.length, 2);
+    });
+
+    QUnit.test('selectAllText', function(assert) {
+        const $list = $('#list').dxList({
+            items: [{ text: 'test 1' }, { text: 'test 2' }],
+            displayExpr: 'text',
+            selectionMode: 'all',
+            showSelectionControls: true,
+            selectAllText: 'custom text'
+        });
+        const list = $list.dxList('instance');
+
+        assert.strictEqual($list.find(`.${LIST_SELECT_ALL_LABEL_CLASS}`).text(), 'custom text');
+
+        list.option('selectAllText', 'Select All');
+        assert.strictEqual($list.find(`.${LIST_SELECT_ALL_LABEL_CLASS}`).text(), 'Select All');
+    });
+
+    QUnit.test('menuItems', function(assert) {
+        const actionSpy = sinon.spy();
+        const menuItems = [{ text: 'action', action: actionSpy }];
+
+        const $list = $('#list').dxList({
+            items: [{ text: 'test 1' }, { text: 'test 2' }],
+            displayExpr: 'text',
+            menuMode: 'context',
+            menuItems
+        });
+
+        let $item = $list.find('.dx-list-item').eq(0);
+        const list = $list.dxList('instance');
+
+        let contextMenuEvent = $.Event('contextmenu', { pointerType: 'mouse' });
+        $item.trigger(contextMenuEvent);
+
+        let $menuItems = $(`.${LIST_CONTEXT_MENUCONTENT_CLASS}`).find('.dx-list-item');
+        assert.strictEqual($menuItems.length, 1, 'items count is correct');
+        $menuItems.eq(0).trigger('dxclick');
+        assert.strictEqual(actionSpy.callCount, 1, 'action is called once');
+
+        list.option('menuItems', []);
+        $item = $list.find('.dx-list-item').eq(0);
+        contextMenuEvent = $.Event('contextmenu', { pointerType: 'mouse' });
+        $item.trigger(contextMenuEvent);
+        $menuItems = $(`.${LIST_CONTEXT_MENUCONTENT_CLASS}`).find('.dx-list-item');
+        assert.strictEqual($menuItems.length, 0, 'items count is correct');
+
+        list.option('menuItems', menuItems);
+        $item = $list.find('.dx-list-item').eq(0);
+        contextMenuEvent = $.Event('contextmenu', { pointerType: 'mouse' });
+        $item.trigger(contextMenuEvent);
+        $menuItems = $(`.${LIST_CONTEXT_MENUCONTENT_CLASS}`).find('.dx-list-item');
+        assert.strictEqual($menuItems.length, 1, 'items count is correct');
+        $menuItems.eq(0).trigger('dxclick');
+        assert.strictEqual(actionSpy.callCount, 2, 'action is called twise');
+    });
```

Comment:
This test case is huge. Can we split it into several test cases?

Reply:
Maybe i can refactor this test

Reply:
See https://github.com/DevExpress/DevExtreme/pull/14751